Notes

AnastasiaBndr opened the pull request:
Зробила два класи, Note та NotesList за аналогом Record та AddressBook, додала можливості видаляти та додавати нотатки та поля keywords та tags. Помилки поки не обробляла окремо а просто робила raise Exception("текст помилки"). Серіалізацію теж не робила

AnastasiaBndr committed:
Notes

Зробила два класи, Note та NotesList за аналогом Record та AddressBook, додала можливості видаляти та додавати нотатки та поля keywords та tags. Помилки поки не обробляла окремо а просто робила raise Exception("текст помилки"). Серіалізацію теж не робила

Nazarovv676 commented on src/note/__init__.py:
```diff
@@ -0,0 +1,7 @@
+from src.note.note import Note
```
У пакетах краще використовувати релятивні імпорти всередині пакетів. Це дасть змогу міняти місце знаходження цих пакетів у проєкті. 

```python
from .note import Note
from .notes_list import NotesList
```

Nazarovv676 commented on addressbook.pkl:
Цей файл завжди буде конфліктним тому що він бінарний (не текстовий). Краще додати його в `.gitignore`

Nazarovv676 commented on src/note/fields/__init__.py:
```diff
@@ -0,0 +1,9 @@
+from src.note.fields.id import Id
```
Тут також імпорти можна зробити релятивними так як це пакет/модуль (є __init__.py)

Nazarovv676 commented on src/note/fields/id.py:
```diff
@@ -0,0 +1,14 @@
+from src.address_book import Field
```
Краще цей клас винести з модуля address_book на рівень вище. Це допоможе пакету notes не залежати від address_book

Nazarovv676 commented on src/note/note.py:
```diff
@@ -0,0 +1,36 @@
+
+
+class Note():
+    def __init__(self, name: str, id: int):
```
id зазвичай йду у першу чергу в параметрах, потім всі інші

Nazarovv676 commented on src/note/note.py:
```diff
@@ -0,0 +1,36 @@
+
+    def add_keyword(self, keyword):
+        if keyword in [keyword.value for keyword in self.keywords]:
+            raise Exception("This keyword exists!")
```
Якщо є якісь частинки коду які треба оновити у майбутньому залишай коментар 
```python
# TODO: use here custom exception
raise Exception()
```

Nazarovv676 commented on src/note/note.py:
```diff
@@ -0,0 +1,36 @@
+
+    def add_tag(self, tag):
+        if tag in [tag.value for tag in self.tags]:
+            raise Exception("This tag exists!")
```
Тут так само

Nazarovv676 commented on src/note/notes_list.py:
```diff
@@ -0,0 +1,24 @@
+from src.note import Note
+
+
+class NotesList(UserDict):
```
Це повинна бути колекція UserList

Nazarovv676 commented on src/note/notes_list.py:
```diff
@@ -0,0 +1,24 @@
+    notes = 0
+
+    def add_note(self, note: Note):
+        self.data[note] = NotesList.notes
```
З UserList тут буде інша логіка

Nazarovv676 commented on src/note/notes_list.py:
```diff
@@ -0,0 +1,24 @@
+        NotesList.notes += 1
+
+    def delete(self, name: str):
+        notes = list(
```
І тут також буде інша логіка

Nazarovv676 commented on src/note/notes_list.py:
```diff
@@ -0,0 +1,24 @@
+            filter(lambda note: note.name.value == name, self.data.keys())
+        )
+        if not len(notes):
+            raise Exception("invalid data format!")
```
Залиш коментар TODO

AnastasiaBndr committed:
Note та Notes

Виправила майже все. Правда у мене чомусь релятивні імпорти не хочуть працювати, тому поки лишила src
Чекни чи все правильно зрозуміла і продовжу далі
add_note поки дуже сира. Написала її так-сяк щоб зрозуміти чи працює воно взагалі, але поки заступорилася на тому, які дані має приймати нотатка. В ТЗ написано що ім'я та опис. Цей опис включає у себе просто текст, а теги та ключові слова ми будемо додавати окремими методами? Чи вони мають бути у цьому описі і потім автоматично присвоюватися полям у конструкторі?

AnastasiaBndr committed:
update imports test

AnastasiaBndr committed:
Merge branch 'main' into Notes

AnastasiaBndr committed:
Notes final update

Впровадила майже весь функціонал. Додані функції додавання, редагування, вадалення та show-notes для показу усіх нотаток і show-note <name> показ нотаток за іменем. Додала для цих функцій месседжі та ще один додатковий експешн

Nazarovv676 commented on src/note/commands/add_note.py:
```diff
@@ -0,0 +1,23 @@
+
+    note = Note(name, description_joined)
+
+    user = book.find(name)
```
Ти можеш прибрати цю перевірку. Нотатки поки що не треба привʼязувати до контактів. Також це не той спосіб привʼязки який ми планували реалізувати. Буде просто додавання нотатки у базу.

```python
@input_error
def add_note(args: list, notes: Notes) -> str:
    name, *description = args
    description_joined = ' '.join(description)

    note = Note(name, description_joined)
    notes.add_note(note)
    return MESSAGES["note_added"]
```

Nazarovv676 commented on src/note/notes.py:
```diff
@@ -0,0 +1,59 @@
+
+    def add_note(self, note: Note):
+        if note not in self.data:
+            if len(self.data) == 0 and Notes.first_open:
```
Чому ти додала цю логіку? pickle не зберігає ids при перезавантаженні програми?

AnastasiaBndr committed:
Notes

InAnadea committed:
Refactor data loading and saving functions; remove notes handling from data storage

AnastasiaBndr committed:
Update main.py

Mariia996 commented on src/note/commands/show_note.py:
```diff
@@ -0,0 +1,24 @@
+    
+    notes = notebook.find_note(name)
+
+    if len(notes)==0:
```
Додай тут відступи спереду і після оператора або використай black на цьому файлі

AnastasiaBndr committed:
Spaces update

AnastasiaBndr committed:
Merge branch 'main' into Notes

AnastasiaBndr committed:
main
